compiler/src/iree/compiler/Codegen/Common/VectorReductionToGPU.cpp

```diff
@@ -40,28 +42,86 @@ static Value allocateGlobalSharedMemory(Location loc, OpBuilder &builder,
   return builder.create<memref::AllocOp>(loc, memrefType);
 }
 
+/// Packs scalar element to it's vector equivalent.
+/// (i.e f16 -> vector<1xf16> and f32 -> vector<1xf32>)
+static Value promoteElementToVector(Location loc, OpBuilder &builder,
+                                    Value input) {
+  VectorType vectorTypeBroadcast = VectorType::get({1}, input.getType());
+  Value vectorInput =
+      builder.create<vector::BroadcastOp>(loc, vectorTypeBroadcast, input);
+  return vectorInput;
+}
+
+/// Packs vector of lower precision into a single 32-bit width element.
+/// (i.e <2xf16> -> i32 and <4xi8> -> i32)
+static Value packVectorToSupportedWidth(Location loc, OpBuilder &builder,
+                                        Value input) {
+  auto vecType = input.getType().cast<VectorType>();
+  Type elementType = vecType.getElementType();
+  assert(vecType.getDimSize(0) * elementType.getIntOrFloatBitWidth() ==
+             kShuffleBitWidth &&
+         "vecSize * vecBitWidth needs to packable into 32-bitwidth.");
+  assert(elementType.isIntOrFloat() &&
```

Comment:
@ThomasRaoux Any thoughts on how to handle asserts with very long variable and getter names. For example, currently this is hitting vecType and elementType is not being used on certain IREE builders/tests. On the other hand if I put in the full getter (i.e input.getType().cast<VectorType>() getElementType(). getIntOrFloatBitWidth() for elementType's bitWidth), feels a bit long and unreadable 😅. Thanks in advance :)